src/uwtools/tests/drivers/test_forecast.py

```diff
@@ -167,6 +168,22 @@ def test_create_namelist_without_base_file(create_namelist_assets):
         assert out_file.read() == expected
 
 
+def test_forecast_run():
+    """
+    Tests that the command to be used to run the forecast executable was built successfully.
+    """
+    hera_expected = "srun --export=ALL test_exec.py"
+    assert hera_expected == FV3Forecast().run("srun", "test_exec.py", "--export=ALL")
+
+    cheyenne_expected = "mpirun -np 4 test_exec.py"
+    assert cheyenne_expected == FV3Forecast().run("mpirun", "test_exec.py", "-np", 4)
+
+    wcoss2_expected = "mpiexec -n 4 -ppn 8 --cpu-bind core -depth 2 test_exec.py"
+    assert wcoss2_expected == FV3Forecast().run(
+        "mpiexec", "test_exec.py", "-n", 4, "-ppn", 8, "--cpu-bind", "core", "-depth", 2
+    )
```

Comment:
After working through implementing our tests, one question that came up was how would address if the user did not provide the executable name. In the case of our tests it would just accept the next arg as the placeholder for the exec_name and would falsely pass the test cases.